packages/service-core/src/storage/mongo/MongoBucketBatch.ts

```diff
@@ -142,45 +147,52 @@ export class MongoBucketBatch extends DisposableObserver<BucketBatchStorageListe
     batch: OperationBatch,
     op_seq: MongoIdSequence
   ): Promise<OperationBatch | null> {
-    // 1. Find sizes of current_data documents, to assist in intelligent batching without
-    // exceeding memory limits.
-    //
-    // A previous attempt tried to do batching by the results of the current_data query
-    // (automatically limited to 48MB(?) per batch by MongoDB). The issue is that it changes
-    // the order of processing, which then becomes really tricky to manage.
-    // This now takes 2+ queries, but doesn't have any issues with order of operations.
-    const sizeLookups: SourceKey[] = batch.batch.map((r) => {
-      return { g: this.group_id, t: r.record.sourceTable.id, k: r.beforeId };
-    });
+    let sizes: Map<string, number> | undefined = undefined;
```

Comment:
Looks like initializing `sizes` with undefined is not necessary and we can revert to the previous initialization?

Reply:
This function specifically uses `sizes = undefined` when `this.storeCurrentData == false` - it does have an effect in some places.

Reply:
Ah I see - hadn't gotten to the OperationBatch change yet 😅